examples/plugins/src/file-size.plugin.integration.test.ts

```diff
@@ -0,0 +1,117 @@
+import { vol } from 'memfs';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { executePlugin } from '@code-pushup/core';
+import { categoryRefSchema, pluginConfigSchema } from '@code-pushup/models';
+import { MEMFS_VOLUME } from '@code-pushup/testing-utils';
+import {
+  PluginOptions,
+  create,
+  audits as fileSizeAudits,
+  recommendedRefs,
+  pluginSlug as slug,
+} from './file-size.plugin';
+
+// Mock file system API's
+vi.mock('fs', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs;
+});
+vi.mock('fs/promises', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs.promises;
+});
+
+const outputDir = MEMFS_VOLUME;
+const projectJson = JSON.stringify(
+  {
+    test: 42,
+    arr: [1, 2, 3],
+    obj: {
+      test: 42,
+    },
+  },
+  null,
+  2,
+);
+const testJs = `
+    const str = 'Hello World'
+    const num = 42;
+    const obj = ${projectJson};
+  `;
+
+describe('create', () => {
+  const baseOptions: PluginOptions = {
```

Comment:
There is no need for a variable, the `directory` option can be passed directly into each test.